MetaMask/metamask-extension
# Update modal scrolling behavior

## **Description**
This PR addresses an issue with the modal's scrolling behavior, which currently relies on the viewport's window rather than the modal's dialog. While this approach(inspired by [this article](https://ux.stackexchange.com/questions/39298/research-on-scrolling-within-modal-web-pros-cons) and [Chakra UI's modal example](https://chakra-ui.com/docs/components/modal/usage)) has its merits, it obscures the `ModalHeader` and modal's footer unless additional styles are applied. 

To ensure these elements are always visible, we've decided to update the modal's scrolling behavior to be internal. This PR implements these changes and adjusts the padding of the `ModalContent`'s dialog, `ModalHeader`, and `ModalBody` accordingly. 

Given the wide usage of modal sub-components throughout the codebase, the current `ModalContent` and `ModalHeader` components have been deprecated to minimize disruption. All instances have been updated to reference these deprecated components, which retain the original padding. Screenshots in the comments confirm no visual changes where stories exist. In 2 cases, the new components have been used to improve scrolling behavior or maximize content area.

Next steps in subsequent PRs:
- Add `ModalFooter` https://github.com/MetaMask/metamask-extension/pull/22155
- Replace deprecated `ModalHeader` and `ModalContent` components

## **Related issues**

Part of: #19413 

## **Manual testing steps**

1. Go to the [storybook build of this PR](https://output.circle-artifacts.com/output/job/cbab845d-9364-4ab4-8997-fb9ae271e293/artifacts/0/storybook/index.html?path=/docs/components-componentlibrary-modal--docs) or pull this branch
2. Search `Modal` in the search bar(make sure its the component-library version)
3. Expand the modal's content to see scrolling behavior

Check all instances of `ModalContent` have been updated
1. Pull this branch 
2. Search `ModalContent,` as this should retrun all instances of the imported version 
3. Ignore the `ui/components/component-library`
4. See 2 updates that have been justified in the comments. There should be no other results

![Screenshot 2023-12-10 at 5 26 22 PM](https://github.com/MetaMask/metamask-extension/assets/8112138/236422de-c9e8-4436-93f6-4138746a0069)


## **Screenshots/Recordings**



### **Before**

https://github.com/MetaMask/metamask-extension/assets/8112138/85a1db7e-086d-4a19-87fa-7676b9f5d528

### **After**

https://github.com/MetaMask/metamask-extension/assets/8112138/55c720fd-ef38-4ac0-9e7c-5ed2803ecad6

Deprecation notices added to the deprecated `ModalContent` and `ModalHeader` components

![Screenshot 2023-12-10 at 5 39 02 PM](https://github.com/MetaMask/metamask-extension/assets/8112138/2284238a-728d-4171-8288-ca56f8cfbb50)
![Screenshot 2023-12-10 at 5 39 08 PM](https://github.com/MetaMask/metamask-extension/assets/8112138/fd0a63aa-5bed-4cc4-a719-5278c8fda552)


## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
